contracts/liquidityMining/KyberFairLaunch.sol

```diff
@@ -337,49 +341,32 @@ contract KyberFairLaunch is IKyberFairLaunch, PermissionAdmin, ReentrancyGuard {
       uint256 reward = uint256(lastAccountedBlock - pool.lastRewardBlock).mul(pool.rewardPerBlock);
       _accRewardPerShare = _accRewardPerShare.add(reward.mul(PRECISION).div(_totalStake));
     }
-    return user.amount.mul(_accRewardPerShare.sub(user.lastRewardPerShare)).div(PRECISION);
-  }
-
-  /**
-  * @dev return latest pool id for a stakeToken
-  * if there is no pool id, return max value
-  */
-  function getLatestPoolId(address _stakeToken) external view returns (uint256) {
-    // if latestTokenPoolId[_stakeToken] is 0 (no pool found),
-    // return a max value which is also uint256(-1)
-    return latestTokenPoolId[_stakeToken] - 1;
-  }
-
-  /**
-  * @dev update pool reward for all pools
-  */
-  function massUpdatePools() public {
-    uint256 length = poolLength;
-    for (uint256 pid = 0; pid < length; ++pid) {
-      updatePoolRewards(pid);
-    }
+    return user.amount.mul(
+      _accRewardPerShare.sub(user.lastRewardPerShare)
+    ).div(PRECISION)
+    .add(user.unclaimedReward);
   }
 
   /**
   * @dev harvest reward from pool for the sender
   * @param _pid: id of the pool
   */
   function harvest(uint256 _pid) public override {
-    require(_pid < poolLength, 'harvest: invalid pool id');
     updatePoolRewards(_pid);
-    _harvest(msg.sender, _pid);
+    _updateUserReward(msg.sender, _pid, true);
   }
 
   /**
   * @dev update reward for one pool
   */
   function updatePoolRewards(uint256 _pid) public override {
+    require(_pid < poolLength, 'invalid pool id');
     PoolInfo storage pool = poolInfo[_pid];
     uint32 lastAccountedBlock = _lastAccountedRewardBlock(_pid);
-    if (lastAccountedBlock <= pool.lastRewardBlock) return;
+    if (lastAccountedBlock <= pool.lastRewardBlock || lastAccountedBlock < pool.startBlock) return;
     uint256 _totalStake = pool.totalStake;
     if (_totalStake == 0) {
-      pool.lastRewardBlock = _safeUint32(block.number);
+      pool.lastRewardBlock = lastAccountedBlock;
       return;
     }
     uint256 reward = uint256(lastAccountedBlock - pool.lastRewardBlock).mul(uint256(pool.rewardPerBlock));
```

Comment:
when you renew, it will update the lastRewardBlock to 110 + latest acc reward per share.
Right after that, it change startBlock, endBlock, lastRewardBlock = startBlock, but keep the acc reward per share.
So in this example:
- a campaign is from block 100 -> 110
- it renews from 120 to 130 -> (call updatePoolReward: lastRewardBlock is 110, accRewardPerShare is updated) -> (startBlock = 120, endBlock = 130, lastRewardBlock = startBlock = 120)
- at block 125, user call updatePoolReward -> only count new reward from block 120 to 125 and update the lastRewardBlock + accRewardPerShare